app/models/factory_log.rb

```diff
@@ -66,4 +71,31 @@ def self.log_set_with_trait(factory_name_and_trait)
   def self.log_set(factory)
     FactoryLog.all.select { |log| log.name == factory }
   end
+
+  # Return all descendants fatcory_log array.
+  # ex:
+  #     [
+  #       [child1, [[grand_son1_1], [grand_son1_2]]],
+  #       [child2, [[grand_son2_1]]],
+  #       [child3]
+  #     ]
+  # TODO: designatable depth
+  def descendants
+    unless children.empty?
+      children.map do |child|
+        [child, child.descendants].compact
+      end
+    end
+  end
+
+  # Retuen all ancestors factory_log array. Index equal depth (from self).
+  def ancestors
+    result = []
+    p = parent
+    result.push(p)
+    while(p = p.parent) do
```

Comment:
Style/SpaceAfterControlKeyword: Use space after control keywords.<br>Style/WhileUntilDo: Do not use `do` with multi-line `while`.